### ISSUE TITLE
New Inclusive Naming Convention Eslint Rule

### ISSUE THREAD

**Issue by bmg817** [eslint/eslint#14831]
**Please describe what the rule should do:**
The new rule should alert developers when a non-inclusive term is being used. Various organizations have already created lists and guidelines for inclusive terminology including [Airbnb](https://github.com/airbnb/inclusion/blob/main/infractions.yml), [Google](https://developers.google.com/style/inclusive-documentation), [Linux](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/Documentation/process/coding-style.rst?id=a5f526ecb075a08c4a082355020166c7fe13ae27), and [IETF](https://tools.ietf.org/id/draft-knodel-terminology-00.html).

**What new ECMAScript feature does this rule relate to?**
Unclear if there are any specific rules. This is particularly related to diversity, equity and inclusion and coding. In general, there's a lot of low hanging fruit that hasn't been addressed historically.

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[x ] Suggests an alternate way of doing something (suggestion)

**Provide 2-3 code examples that this rule will warn about:**

```js
const master = ...  // replace with: main, primary, or leader 
const slave = ... // replace with: secondary, or follower
```

**Why should this rule be included in ESLint (instead of a plugin)?**
To expedite migration to more inclusive terminology in the software engineering field. This is an urgent need given we still have a field where people of minority communities and women don't feel welcome. This is low hanging fruit that can make a major difference.

**Are you willing to submit a pull request to implement this rule?**
Yes

**Comment by bmg817** [comment]
Just kidding. Found a way to do this with the powerful `no-restricted-syntax` rule configurations. Will close now.
